common/types/hashes.go

```diff
@@ -177,7 +177,16 @@ func CalcHash32(data []byte) Hash32 {
 func CalcATXHash32(atx *ActivationTx) Hash32 {
 	bytes, err := codec.Encode(atx)
 	if err != nil {
-		panic("could not Serialize atx")
+		panic("could not serialize ATX")
+	}
+	return CalcHash32(bytes)
+}
+
+// CalcProposalHash32 returns the 32-byte sha256 sum of serialization of the given Proposal.
+func CalcProposalHash32(proposal *Proposal) Hash32 {
```

Comment:
it looks to me like there could be one generic method to compute hash from scale.Encodable

Reply:
Created a function that takes a `scale.Encodable` for `Hash32` calculation of an object.